Gems/MotionMatching/Code/Source/TrajectoryQuery.cpp

```diff
@@ -21,39 +21,100 @@ namespace EMotionFX::MotionMatching
         return displacement;
     }
 
+    void TrajectoryQuery::PredictFutureTrajectory(const ActorInstance* actorInstance,
+        const FeatureTrajectory* trajectoryFeature,
+        const AZ::Vector3& targetPos,
+        [[maybe_unused]] const AZ::Vector3& targetFacingDir)
+    {
+        const AZ::Vector3 actorInstanceWorldPosition = actorInstance->GetWorldSpaceTransform().m_position;
+        const AZ::Quaternion actorInstanceWorldRotation = actorInstance->GetWorldSpaceTransform().m_rotation;
+        const AZ::Vector3 actorInstanceToTarget = (targetPos - actorInstanceWorldPosition);
+
+        const size_t numFutureSamples = trajectoryFeature->GetNumFutureSamples();
+        const float numSections = aznumeric_cast<float>(numFutureSamples-1);
+
+        float linearDisplacementPerSample = 0.0f;
+        AZ::Quaternion targetFacingDirQuat = actorInstanceWorldRotation;
+        if (!actorInstanceToTarget.IsClose(AZ::Vector3::CreateZero(), m_deadZone))
+        {
+            // Calculate the desired linear velocity from the current position to the target position based on the trajectory future time range.
+            AZ_Assert(trajectoryFeature->GetFutureTimeRange() > AZ::Constants::FloatEpsilon, "Trajectory feature future time range is too small.");
+            const float velocity = actorInstanceToTarget.GetLength() / trajectoryFeature->GetFutureTimeRange();
+
+            linearDisplacementPerSample = (velocity / numSections);
+
+            // Use the direction from the current actor instance position to the target as the target facing direction
+            // and convert the direction vector to a quaternion.
+            targetFacingDirQuat = AZ::Quaternion::CreateShortestArc(trajectoryFeature->GetFacingAxisDir(), actorInstanceToTarget);
+        }
+
+        // Set the first control point to the current position and facing direction.
+        m_futureControlPoints[0].m_position = actorInstanceWorldPosition;
+        m_futureControlPoints[0].m_facingDirection = actorInstanceWorldRotation.TransformVector(trajectoryFeature->GetFacingAxisDir());
+
+        for (size_t i = 1; i < numFutureSamples; ++i)
+        {
+            const float t = aznumeric_cast<float>(i) / numSections;
```

Comment:
nit: Should we use `time` here instead of just `t`? (I think the coding standards to prefer avoiding single letter variable names if possible)

Reply:
I am all on the side of descriptive variable names, but in here this is just the function parameter for a parameterized function and the standard for that is to use t as far as I am aware. This is not related to a timing value.